lib/samples/densify_and_generalize_geometry/densify_and_generalize_geometry_sample.dart

```diff
@@ -0,0 +1,364 @@
+  const DensifyAndGeneralizeGeometrySample({super.key});
+
+  @override
+  State<DensifyAndGeneralizeGeometrySample> createState() =>
+      _DensifyAndGeneralizeGeometrySampleState();
+}
+
+class _DensifyAndGeneralizeGeometrySampleState
+    extends State<DensifyAndGeneralizeGeometrySample> with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // Declare a polyline geometry representing the ship's route.
+  late final Polyline _originalPolyline;
+  // Declare a graphic for displaying the points of the resultant geometry.
+  late final Graphic _resultPointsGraphic;
+  // Declare a graphic for displaying the lines of the resultant geometry.
+  late final Graphic _resultPolylineGraphic;
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // A flag for when the settings bottom sheet is visible.
+  var _settingsVisible = false;
+  // A flag for whether to generalize the geometry.
+  var _generalize = false;
+  // The maximum deviation for the generalization.
+  var _maxDeviation = 10.0;
+  // A flag for whether to densify the geometry.
+  var _densify = false;
+  // The maximum segment length for the densification.
+  var _maxSegmentLength = 100.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    // A button to show the Geometry Settings bottom sheet.
+                    ElevatedButton(
+                      onPressed: () => setState(() => _settingsVisible = true),
+                      child: const Text('Geometry Settings'),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+      // The Geometry Settings bottom sheet.
+      bottomSheet: _settingsVisible ? buildSettings(context) : null,
+    );
+  }
+
+  // The build method for the Geometry Settings bottom sheet.
+  Widget buildSettings(BuildContext context) {
```

Comment:
Probably it would need rounded corners and maybe a drag handle to swoosh it closed.

I considered the DraggableScrollableSheet class that comes with Flutter, but the "scrollable" part is wrong for what we're doing here. https://api.flutter.dev/flutter/widgets/DraggableScrollableSheet-class.html